backend/koofr/koofr.go

```diff
@@ -0,0 +1,461 @@
+package koofr
+
+import (
+	"encoding/base64"
+	"errors"
+	"fmt"
+	"io"
+	"net/http"
+	"path"
+	"strings"
+	"time"
+
+	"github.com/ncw/rclone/fs"
+	"github.com/ncw/rclone/fs/config/configmap"
+	"github.com/ncw/rclone/fs/config/configstruct"
+	"github.com/ncw/rclone/fs/config/obscure"
+	"github.com/ncw/rclone/fs/hash"
+
+	httpclient "github.com/koofr/go-httpclient"
+	koofrclient "github.com/koofr/go-koofrclient"
+)
+
+func init() {
+	fs.Register(&fs.RegInfo{
+		Name:        "koofr",
+		Description: "Koofr",
+		NewFs:       NewFs,
+		Options: []fs.Option{
+			{
+				Name:     "baseurl",
+				Help:     "Base URL of the Koofr API to connect to",
+				Default:  "https://app.koofr.net",
+				Required: true,
+				Advanced: true,
+			}, {
+				Name:     "mountid",
+				Help:     "Mount ID of the mount to use. If omitted, the primary mount is used.",
+				Required: false,
+				Default:  "",
+				Advanced: true,
+			}, {
+				Name:     "user",
+				Help:     "Your Koofr user name",
+				Required: true,
+			}, {
+				Name:       "password",
+				Help:       "Your Koofr password for rclone (generate one at https://app.koofr.net/app/admin/preferences/password)",
+				IsPassword: true,
+				Required:   true,
+			},
+		},
+	})
+}
+
+type options struct {
+	BaseURL  string `config:"baseurl"`
+	MountID  string `config:"mountid"`
+	User     string `config:"user"`
+	Password string `config:"password"`
+}
+
+type koofrfs struct {
+	name     string
+	mountID  string
+	root     string
+	opt      options
+	features *fs.Features
+	client   *koofrclient.KoofrClient
+}
+
+type object struct {
```

Comment:
And this called `Object`

Those 3 things are public so the backends can be used without the rest of the rclone framework.

Reply:
lol. I have only unexported the types after the golangci linter complained extensively about undocumented exported stuff, as I hate the noise of warnings during builds. :D

you really should add something to the CONTRIBUTING.md about ignoring the damn linter. ;)

kudos for the PR submission policy, though.